### ISSUE TITLE
Arm64 -- qemu-kvm-device-usb-redirect package is missing on Arm64 platform

### ISSUE THREAD

**Issue by zhlhahaha** [kubevirt/kubevirt#10208]
**What happened**:
As there is no rpm package `qemu-kvm-device-usb-redirect` in https://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/, clientPassthrough does not works on Arm64 platform.
https://github.com/kubevirt/kubevirt/issues/10200

**What you expected to happen**:
clientPassthrough works on Arm64 platform

**Additional context**:
Shall we request to add `qemu-kvm-device-usb-redirect` in https://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/?

**Comment by zhlhahaha** [comment]
cc: @rmohr @xpivarc @andreabolognani

**Comment by zhlhahaha** [comment]
> My understanding is that there are no plans to support ARM as of now. 

Hi, @xpivarc , thanks for replying.

From my understanding, here are two solution for USB mapping,
1. use USB redirection
2. use USB host pass through

In terms of the first solution, as we are lack of the rpm package `qemu-kvm-device-usb-redirect` on Arm64, so the qemu-kvm just do not support USB redirect. From my point of view, if we have this package and install it in `virt-launcher`, USB redirection should work.

For the second solution, the PR of @victortoso provide a way to solve it but it seems that a new PR is in process https://github.com/kubevirt/kubevirt/pull/10015#issuecomment-1658050545. As we have Arm64 version `qemu-kvm-device-usb-host` in https://mirror.stream.centos.org/9-stream/AppStream/aarch64/os/Packages/, so USB host pass through should works on Arm64 on qemu level. I will try @victortoso ' PR on Arm64, once it ready.

**Comment by victortoso** [comment]
> My understanding is that there are no plans to support ARM as of now.

There are no plans but, the issue is valid in case someone wants to have a go in enabling it.

I'm not sure why [Fedora](https://koji.fedoraproject.org/koji/buildinfo?buildID=2263283) and centos do not have `qemu-kvm-device-usb-redirect` built enabled by default on arm. I think it should work as I recall SPICE users on raspberry pi.

> if we have this package and install it in virt-launcher, USB redirection should work.

Yes and it shouldn't be too hard to test it as well.

**Comment by andreabolognani** [comment]
> I'm not sure why [Fedora](https://koji.fedoraproject.org/koji/buildinfo?buildID=2263283) and centos do not have `qemu-kvm-device-usb-redirect` built enabled by default on arm.

You're looking at the wrong Fedora package.

usbredir support is explicitly disabled on !x86 in the [CentOS Stream package](https://gitlab.com/redhat/centos-stream/rpms/qemu-kvm/-/blob/d10ea1f259020b4281e54e5166dba0e2fcad736b/qemu-kvm.spec#L61-63), and that reflects on what's available in the CentOS Stream mirror.

The [Fedora package](https://src.fedoraproject.org/rpms/qemu/blob/rawhide/f/qemu.spec) doesn't explicitly disable usbredir support on any architecture, so [Fedora builds](https://koji.fedoraproject.org/koji/buildinfo?buildID=2251777) have it enabled on aarch64 as well as ppc64le and s390x.

I have no idea why support is limited to x86 in CentOS Stream. I have to assume that's due to lack of interest in the feature on other architectures so far, as well as the desire to keep the footprint minimal and maintainable for a small team. You could try asking for the feature to be enabled, although there's no guarantee that it will actually happen. If the KubeVirt maintainers claim interest in it, that could probably help your chances :)

**Comment by zhlhahaha** [comment]
Here is the bugzilla links https://bugzilla.redhat.com/show_bug.cgi?id=2231275

**Comment by andreabolognani** [comment]
@zhlhahaha thanks for opening the ticket.

I think it would be useful if you could make the case (on Bugzilla) for *why* you think the device should be enabled. Is it just because x86 has it? Is there a concrete usage scenario for USB redirect on Arm?

Note that I know next to nothing about the topic, but I expect that this is the kind of questions that the maintainers will have to ask in order to make a call. Might as well provide the information ahead of time, which will hopefully make the process smoother :)

**Comment by yashkmankad** [comment]
Hi @zhlhahaha , just picked up this bugzilla. Discussing internally and will work on it in the coming week.

**Comment by zhlhahaha** [comment]
> Discussing

Thanks @yashkmankad , feel free to tell me if I can provide any help.

**Comment by zhlhahaha** [comment]
> Hi @zhlhahaha , just picked up this bugzilla. Discussing internally and will work on it in the coming week.

 Hi @yashkmankad , are there any update on this issue?

**Comment by yashkmankad** [comment]
Hi @zhlhahaha, thanks for the ping. 
Unfortunately hasn't moved ahead yet, following up.

**Comment by andreabolognani** [comment]
@zhlhahaha the package is available from CentOS Stream 9 now. I'll look into picking it up next time I update the virtualization packages, which will hopefully be soon... I'm just waiting for the libvirt 10.0.0 build to make it to the mirrors.

/remove-lifecycle stale

**Comment by zhlhahaha** [comment]
> @zhlhahaha the package is available from CentOS Stream 9 now. I'll look into picking it up next time I update the virtualization packages, which will hopefully be soon... I'm just waiting for the libvirt 10.0.0 build to make it to the mirrors.
> 
> /remove-lifecycle stale

Thanks for informing! @andreabolognani

**Comment by andreabolognani** [comment]
/remove-lifecycle stale

Still relevant. We will be able to address it once #11256 lands, which is hopefully going to be soon.

**Comment by andreabolognani** [comment]
This shouldn't have been closed, and I'm not sure why it has. @zhlhahaha, is it possible for you to reopen it? I've created #11869 which actually addresses it in the meantime.
